frontends/ast/genrtlil.cc

```diff
@@ -1720,16 +1720,31 @@ RTLIL::SigSpec AstNode::genRTLIL(int width_hint, bool sign_hint)
 					continue;
 				}
 				if (child->type == AST_ARGUMENT) {
-					RTLIL::SigSpec sig;
-					if (child->children.size() > 0)
-						sig = child->children[0]->genRTLIL();
+					RTLIL::IdString port_id;
 					if (child->str.size() == 0) {
 						char buf[100];
 						snprintf(buf, 100, "$%d", ++port_counter);
-						cell->setPort(buf, sig);
+						port_id = buf;
 					} else {
-						cell->setPort(child->str, sig);
+						port_id = child->str;
+					}
+					RTLIL::SigSpec sig;
+					if (child->children.size() > 0) {
+						AstNode *arg = child->children[0];
+						if (arg->type == AST_CONSTANT && arg->is_signed) {
```

Comment:
I'm not convinced that this covers every possible case. I think it would be more robust to add an indirect wire each time the node connected to the port is signed and the output of `genRTLIL` is not a wire. And maybe add an assert for the case where it is a wire but the wire is not signed, since that seems like it should never happen.